shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,227 @@
+}
+
+pub fn parse_json_for_seq_num(json_objects: Vec<Value>) -> Result<u64> {
+    let mut seq_number_string = "";
+    for object in &json_objects {
+        if object["type"]["name"] == "DiemAccount" {
+            seq_number_string = object["value"]["sequence_number"]
+                .as_str()
+                .ok_or_else(|| anyhow!("Invalid sequence number string"))?;
+            break;
+        };
+    }
+    let seq_number: u64 = seq_number_string.parse()?;
+    Ok(seq_number)
+}
+
+fn write_into<W>(writer: &mut W, json: &serde_json::Value) -> Result<()>
+where
+    W: Write,
+{
+    write!(writer, "{}", json)?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use crate::transactions::{parse_json_for_seq_num, write_into, write_out_txns};
+    use serde_json::{json, map::Map, Value};
+
+    #[test]
+    fn test_parse_json_for_seq_num() {
+        let mut name_map: Map<String, Value> = Map::new();
+        name_map.insert("name".parse().unwrap(), Value::from("DiemAccount"));
+
+        let mut seq_num_map: Map<String, Value> = Map::new();
+        seq_num_map.insert("sequence_number".parse().unwrap(), Value::from("3"));
+
+        let mut overall_map: Map<String, Value> = Map::new();
+        overall_map.insert("type".parse().unwrap(), Value::from(name_map));
+        overall_map.insert("value".parse().unwrap(), Value::from(seq_num_map));
+
+        let json_obj: Vec<Value> = vec![Value::Object(overall_map)];
+        let ret_seq_num = parse_json_for_seq_num(json_obj).unwrap();
+        assert_eq!(ret_seq_num, 3);
+    }
+
+    #[test]
+    fn test_write_into() {
+        let mut stdout = Vec::new();
+        let val = Value::from("test");
+        write_into(&mut stdout, &val).unwrap();
+        assert_eq!(r#""test""#, String::from_utf8(stdout).unwrap());
+    }
+
+    #[test]
+    fn test_write_out_txns_last_txn_sqn_num_gt_old_seq_num() {
+        let json = json!([
+            "already_printed_out_txn",
+            "txn_that_hasn't_been_printed_out_yet"
+        ]);
+        let mut stdout = Vec::new();
+        let next_old_seq_num = write_out_txns(json, 6, 5, &mut stdout).unwrap();
+        assert_eq!(
+            String::from_utf8(stdout).unwrap(),
+            r#""txn_that_hasn't_been_printed_out_yet""#
+        );
+        assert_eq!(next_old_seq_num, 6);
+    }
+
+    #[test]
+    fn test_write_out_txns_last_txn_sqn_num_eq_zero() {
+        let json = json!([
+            "first_txn_that_hasn't_been_printed_out_yet",
+            "txn_that_hasn't_been_printed_out_yet"
+        ]);
+        let mut stdout = Vec::new();
+        let next_old_seq_num = write_out_txns(json, 0, 0, &mut stdout).unwrap();
+        assert_eq!(
+            String::from_utf8(stdout).unwrap(),
+            r#""first_txn_that_hasn't_been_printed_out_yet""txn_that_hasn't_been_printed_out_yet""#
+        );
+        assert_eq!(next_old_seq_num, 0);
+    }
+
+    #[test]
+    fn test_write_out_txns_last_txn_sqn_num_eq_old_seq_num() {
```

Comment:
When you have many small tests like this with similar boiler place, check out table driven tests: 

https://github.com/dimroc/diem/blob/8fab7b7f2e131bdcfad475b0509c636e5e1883bf/crypto/crypto/src/unit_tests/compat_test.rs#L10-L17